### PR TITLE
[NEW] Support message read receipt

### LINKED ISSUE #1331: [NEW] Support "Message read receipt"
## General information

- PR that implements it in our back-end: https://github.com/RocketChat/Rocket.Chat/pull/9717
- Property of the message to look into: `unread` (Boolean)
- Settings to be verified if feature is enabled:
  - `Message_Read_Receipt_Enabled`
  - `Message_Read_Receipt_Store_Users`

## Assets

[Check.zip](https://github.com/RocketChat/Rocket.Chat.iOS/files/1748153/Check.zip)

Color when **not read** by everyone: `#999999`
Color when **read** by everyone: `#1D74F5`

## Description

When the setting `Message_Read_Receipt_Enabled` is enabled, the Message objects are going to have a property called `unread`. The app needs to map this value and show a "check" information on the right side of the timestamp of the message. When feature is disable, this check image is not displayed.

If the setting `Message_Read_Receipt_Store_Users ` is enabled, when user long press the message, we're going to have a new option called **Information** that will display only the message in a new screen and below the message there will be a list of users that read the message.

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

Closes #1331

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChatControllerMessageCellProtocol.swift [hunk 1]
```diff
@@ -72,6 +72,40 @@ extension ChatViewController: ChatMessageCellProtocol, UserActionSheetPresenter
...
+        guard let messageId = message.identifier else {
+            return
+        }
+
+        let controller = ReadReceiptListViewController()
+        controller.modalPresentationStyle = .popover
+        _ = controller.view
+
+        let api = API.current()
+        api?.fetch(ReadReceiptsRequest(messageId: messageId)) { response in
```
Comment (filipealva) [inline Rocket.Chat/Controllers/Chat/ChatControllerMessageCellProtocol.swift]: Isn't it better to do this call inside of `ReadReceiptListViewController`?

Comment (rafaelks) [reply Rocket.Chat/Controllers/Chat/ChatControllerMessageCellProtocol.swift]: I agree here! 👍

Comment (cardoso) [reply Rocket.Chat/Controllers/Chat/ChatControllerMessageCellProtocol.swift]: Done :+1:

**Comment by cardoso** [comment]
Changes done! Thanks for the feedbacks 👏

File: Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings [hunk 2]
```diff
@@ -149,6 +149,11 @@
 // Notifications
 "notifications.action.reply" = "Reply";
 
+// Read Receipts
+"chat.read_receipt_list.title" = "Lido por"; //TODO
+"chat.read_receipt_list.title.loading" = "Loading..."; //TODO
+"chat.read_receipt_list.title.empty" = "Lido por ninguém"; //TODO
```
Comment (rafaelks) [inline Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings]: Not todo

File: Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings [hunk 3]
```diff
@@ -149,6 +149,11 @@
 // Notifications
 "notifications.action.reply" = "Reply";
 
+// Read Receipts
+"chat.read_receipt_list.title" = "Lido por"; //TODO
+"chat.read_receipt_list.title.loading" = "Loading..."; //TODO
```
Comment (rafaelks) [inline Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings]: Not todo

File: Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings [hunk 4]
```diff
@@ -160,6 +165,7 @@
 "chat.unread_separator" = "novas mensagens";
 
 // Chat Message Actions
+"chat.message.actions.info" = "Informação"; //TODO
```
Comment (rafaelks) [inline Rocket.Chat/Resources/pt-BR.lproj/Localizable.strings]: Not todo

**Pull request merged**